apps/els_lsp/src/els_refactorerl_diagnostics.erl

```diff
@@ -0,0 +1,94 @@
+%%==============================================================================
+%% RefactorErl Diagnostics
+%%==============================================================================
+-module(els_refactorerl_diagnostics).
+
+%%==============================================================================
+%% Behaviours
+%%==============================================================================
+-behaviour(els_diagnostics).
+
+%%==============================================================================
+%% Exports
+%%==============================================================================
+-export([ is_default/0
+        , run/1
+        , source/0
+        ]).
+
+%%==============================================================================
+%% Includes & Defines
+%%==============================================================================
+-include("els_lsp.hrl").
+
+%%==============================================================================
+%% Types
+%%==============================================================================
+-type refactorerl_diagnostic_id() :: {atom(), [char()], [char()], [char()]}.
+-type refactorerl_query() :: [char()].
+
+%%==============================================================================
+%% Callback Functions
+%%==============================================================================
+
+-spec is_default() -> boolean().
+is_default() ->
+  false.
+
+-spec run(uri()) -> [els_diagnostics:diagnostic()].
+run(Uri) ->
+  case filename:extension(Uri) of
+    <<".erl">> ->
+      case els_refactorerl_utils:referl_node() of
+        {error, _} ->
+          [];
+        {ok, _} ->
+          case els_refactorerl_utils:add(Uri) of
+            error ->
+              [];
+            ok ->
+              FileName = filename:basename(binary_to_list(els_uri:path(Uri))),
+              Module = list_to_atom(filename:rootname(FileName)),
+              Diagnostics = refactorerl_diagnostics(),
```

Comment:
Nit: the name is a bit misleading here: it suggests the function is returning a list of diagnostics, but in reality it returns a list of _identifiers_ (which actually looks more like _specs_, of the form `{id(), description(), query()}`).

Reply:
I also made it configurable. Maybe `enabled_diagnositcs()` can work out?